internal/receiver/smartagentreceiver/log_test.go

```diff
@@ -0,0 +1,75 @@
+package smartagentreceiver
+
+import (
+	"io/ioutil"
+	"sync"
+	"testing"
+
+	"github.com/sirupsen/logrus"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zaptest"
+)
+
+func newLogRedirector(t *testing.T) *logRedirector {
+	return &logRedirector{
+		redirects:         make(map[srcLogger][]dstLogger),
+		mu:                sync.Mutex{},
+		dstCatchallLogger: zaptest.NewLogger(t),
```

Comment:
zaptest provides observers to confirm that logged statements are emitted as expected.  Would you be able to add a more integration-level test to confirm that `Fire()` and redirection work with multiple loggers and logrus?